src/wp-includes/comment-template.php

```diff
@@ -2008,20 +2034,31 @@ function comment_form_title( $no_reply_text = false, $reply_text = false, $link_
 
 	$reply_to_id = isset( $_GET['replytocom'] ) ? (int) $_GET['replytocom'] : 0;
 
-	if ( 0 == $reply_to_id ) {
+	if ( ! $post_id || 0 === $reply_to_id ) {
 		echo $no_reply_text;
-	} else {
-		// Sets the global so that template tags can be used in the comment form.
-		$comment = get_comment( $reply_to_id );
+		return;
+	}
 
-		if ( $link_to_parent ) {
-			$author = '<a href="#comment-' . get_comment_ID() . '">' . get_comment_author( $comment ) . '</a>';
-		} else {
-			$author = get_comment_author( $comment );
-		}
+	// Sets the global so that template tags can be used in the comment form.
+	$comment = get_comment( $reply_to_id );
 
-		printf( $reply_text, $author );
+	if ( ! $comment ) {
```

Comment:
`get_comment()` will return an instance of the object or `null`. Instead of doing a falsey check here, I'd suggest a more direct check of the opposite expected:

```suggestion
	if ( ! $comment instanceof WP_Comment ) {
```

one could argue an exact check for the failed response would checking for `null`:
```php
	if ( null === $comment ) {
```

Here, either is okay. But be consistent and apply the same check implementation in all instances.

Reply:
I agree that checking the response of `get_comment()` should be reviewed. I used `if ( ! $comment )` as it's consistent with the rest of the codebase - there are 23 other instances of `if ( ! $comment )` in the codebase and none that check for `instanceof` or `null === $comment`, both of which I agree are better.

See [this discussion](https://wordpress.slack.com/archives/C02RQBWTW/p1630738445035900) on Slack where I raised this with @jrfnl. The discussion led to a suggestion that a blessed task be created to audit `get_comment()` calls and response checks. I haven't had a chance to follow up on this as of yet - and I remembered an embarrassingly long time after the discussion that I can't actually create a blessed task.

I'm happy to commit the `if ( ! $comment instanceof WP_Comment )` suggestion above for this patch. Is it better to make this change here before discussing the other uses, or submit this as consistent with the current codebase and update all references if a ticket discussion leads us that way?

Reply:
@costdev if you create the ticket, just ping me with the link and I'll turn it into a blessed task for you ;-)

Reply:
@jrnfl [Done!](https://core.trac.wordpress.org/ticket/54149)

Reply:
And Done ;-)